.ci/helper_shellchecks.sh

```diff
@@ -10,8 +10,13 @@ mapfile -t shellscript_locations < <({ git grep -lE '^#!(/usr)?/bin/(env )?(bash
 SHELLSCRIPT_ERROR=0
 
 for shellscript in "${shellscript_locations[@]}"; do
-    echo -e "${ANSI_GREEN}Running shellcheck on ${shellscript}"
-    shellcheck "${shellscript}" || SHELLSCRIPT_ERROR=1
+    # These two scripts are running on the device, so they should use pure sh
```

Comment:
If there is a valid reason to skip a check you should skip the check, not ShellCheck itself. You can do this by putting `# shellcheck disable=SC1234` on the preceding line.

However, I don't see much that should be skipped at a glance. The overly complex piping with `ps` and `grep` smells. In other words, it should probably be rewritten regardless even if the error message technically doesn't apply because we already know the PID. Something like `kill -s 0 $PID` should do the trick much better. None of this has anything to do with "pure sh syntax" as all these commands (`ps`, `pgrep` and `kill`) are provided by BusyBox so I'm not quite sure where you're going with that particular statement regardless. ;-)

`if [ $? -eq 0 ]` is not a good idea because it's easy to accidentally start checking the exit code for something else. What you should write instead is something like this:

```bash
    if EXIT_CODE=$(kill -s 0 $PID >/dev/null 2>&1); then
        # Unblock f:read().
        echo
    else
        wait $JOB_ID
        if [ "$EXIT_CODE" -eq "255" ]; then
            TIMEOUT="true"
        #etc
```
That way you'll never quietly introduce failures of this sort:
```bash
ps -p $JOB_ID | grep $JOB_ID >/dev/null 2>&1
echo "w00t what an awesome program this is"
if [ $? -eq 0 ]
#oops, we just checked echo
```

Finally, the warning about it being interpreted as a literal is a valid warning. It's just that we want it as a literal. So you can either make it disappear by disabling the error check, or you can quote it in various ways, which I happen to find clearer and therefore prefer style-wise. But that's personal.

```bash
        echo "return { \
            result = $EXIT_CODE, \
            timeout = $TIMEOUT, \
            bad_command = $BADCOMMAND, \
            }"
```

```bash
        echo "return {" \
            result = $EXIT_CODE, \
            timeout = $TIMEOUT, \
            bad_command = $BADCOMMAND, \
            "}"
```

```bash
        echo "return {" \
            "result = $EXIT_CODE," \
            "timeout = $TIMEOUT," \
            "bad_command = $BADCOMMAND," \
            "}"
```